doc/adr/0011-tsp-analysis-api.md

```diff
@@ -178,100 +183,175 @@ sequenceDiagram
     client->>client: Refresh UI
 ```
 
-### Configuration service per experiment
+### Configure customizable outputs 
+
+Outputs might accept input parameters that will configure the analysis used to create the output. Such customization can create new outputs or provide additional query parameters. The TSP api will allow the back-end to define configuration types for that. Each configuration type will use `CustomizationTypeSource` to define what parameters need to be provided to create a new and derived output. The derived output descriptor will store the configuration details. Derived outputs can be deleted using a DELETE operation.
 
-For this data provider service will be augmented for managing configurations per experiment.
+    GET /experiments/{expUUID}/outputs/{outputId}/configTypes
+    Returns a list of configuration source types: typeId, name, description, scope, expected parameter descriptors (e.g. "path" for file path)
 
-    GET experiments/{expUUID}/outputs/config
-        returns a map typeId -> list of configuration descriptors of existing configurations on server
-    POST experiments/{expUUID}/outputs/config
-        Assign configuration to an experiment using typeId and configId from above.
-        Returns configuration descriptor and list of data provider descriptors (if available)
-    DELETE experiments/{expUUID}/outputs/config/{configId}
-        Removes a configuration from an experiment
+    GET /experiments/{expUUID}/outputs/{outputId}/configTypes/{typeId}
+    Returns a single configuration source type for given typeId: typeId, name, description, scope, expected parameter descriptors (e.g. "path" for file path)
 
-#### Sequence: Create configuration instance for an experiment
+    POST /experiments/{expUUID}/outputs/{outputId}
+    Create derived output. It returns a derived output descriptor with a unique Derived Output ID. It will contain the configuration used to create to allow manual reuse, as well as, the parent output ID. The parameter must include typeId and config adhering to the configuration type referenced by typeId.
 
-The following illustrates the sequence of events and messages to create an configuration instance for a given type and experiment. It uses the Trace Compass In-And-Out as example. Note, that the configuration is provided using a file.
+    DELETE /experiments/{expUUID}/outputs/{outputId}/{derivedOutputId}
```

Comment:
Not sure what are the pros & cons of this instead of directly?
DELETE /experiments/{expUUID}/outputs/{derivedOutputId}

Reply:
Good question. I think it's more understandable because POST on an endpoint (/experiments/{expUUID}/outputs/{outputId}) creates a new resource identified on the derivedOutputId and delete on that endpoint/derivedOutputId is to delete. If we put on a different endpoint it might not be as clear. 

I'll keep it for now.